src/modules/partition/core/PartitionLayout.cpp

```diff
@@ -28,17 +31,32 @@
 #include <kpmcore/core/partition.h>
 #include <kpmcore/fs/filesystem.h>
 
+static int
+getDefaultFileSystemType()
+{
+    Calamares::GlobalStorage* gs = Calamares::JobQueue::instance()->globalStorage();
+    int defaultFs = FileSystem::typeForName( gs->value( "defaultFileSystemType" ).toString() );
```

Comment:
This looks kind of fragile: it's dependent on that key being already-inserted into the GlobalStorage, It **is**, in the current code-setup, by PartitionViewStep::setConfigurationMap(), but ..

Reply:
Right, I'll check it and fall back to ext4 if the key is missing